### PR TITLE
Framework: Reduxify help links

### PR DESCRIPTION
**Pull request by tyxla**
The original help search uses a legacy help links Flux store. This PR aims to get rid of that legacy store and reduxify it in order to be consistent with our recent code best practices.

Since the original Flux store was very small and the new Redux implementation is rather small too, I thought we could do this in a single PR. A list of the changes this PR introduces is:

* Add a simple Redux implementation for help links (actions, data layer, reducer, selector and query component)
* Update the existing `HelpSearch` component to use the Redux implementation.
* Remove the usage of an undefined `search` prop in `HelpSearch`.
* Sort some imports and props in `HelpSearch`.
* Remove unnecessary `displayName` in `HelpSearch`.
* Destructure `translate` for shorter expressions in `HelpSearch`.
* Completely remove the no longer used legacy `lib/help-search` Flux store.

This PR also introduces a small improvement over the previous implementation. Previously, when we search for something and then we erase our search query, an useless network request was issued with an empty search query. This will no longer happen, as our query component will guard for empty search queries and will only issue a request when a search query is present.

You might notice the PR lacks any state tests - this is intentional - any tests would only duplicate the existing code that they test, which we're consciously trying to avoid.

To test:
* Checkout this branch
* Go to http://calypso.localhost:3000/help
* Play with the search form at the top of the page.
* Verify the search form works like it did before, and there are no behavioral or visual changes, and no issues/side effects introduced.

### CODE REVIEW

File: client/me/help/help-search/index.jsx [hunk 1]
```diff
@@ -97,23 +86,23 @@ export class HelpSearch extends React.PureComponent {
 		return (
 			<div>
 				<HelpResults
-					header={ this.props.translate( 'WordPress.com Documentation' ) }
+					footer={ translate( 'See more from WordPress.com Documentation…' ) }
+					header={ translate( 'WordPress.com Documentation' ) }
 					helpLinks={ helpLinks.wordpress_support_links }
```
Comment (sirreal) [inline client/me/help/help-search/index.jsx]: These unguarded property accesses concern me. I know they're not new, but we don't have added safety of a transformer in `fromApi` or any additional safety around Redux state.

File: client/state/help/reducer.js [hunk 2]
```diff
@@ -21,9 +20,21 @@ export const selectedSiteId = createReducer( null, {
...
+/**
+ * Responsible for the help search results links
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const links = createReducer( [], {
+	[ HELP_LINKS_RECEIVE ]: ( state, action ) => action.helpLinks,
+} );
```
Comment (sirreal) [inline client/state/help/reducer.js]: How about a classic `function` with a `switch`? There's some overhead to createReducer and I don't see any benefit here.

Comment (tyxla) [reply client/state/help/reducer.js]: Good call, done in https://github.com/Automattic/wp-calypso/pull/27020/commits/09aa12fb92261ae8b12bbff5437c8f998bfd0941.

File: client/me/help/help-search/index.jsx (hunk 1 above)
Comment (tyxla) [reply client/me/help/help-search/index.jsx]: Neither we had any additional safety before 🤷‍♂️ 

But if you insist, I'm happy to add some additional safety (like `_.get()` for example).

File: client/me/help/help-search/index.jsx [hunk 3]
```diff
@@ -12,40 +11,30 @@ import { localize } from 'i18n-calypso';
...
-
-	refreshHelpLinks = () => this.setState( { helpLinks: HelpSearchStore.getHelpLinks() } );
-
 	onSearch = searchQuery => {
-		this.setState( { helpLinks: {}, searchQuery: searchQuery } );
-		this.props.fetchSearchResults( searchQuery );
+		this.setState( {
+			searchQuery,
+		} );
+		recordTracksEvent( 'calypso_help_search', { query: searchQuery } );
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/me/help/help-search/index.jsx]: No; we've already been tracking it - see https://github.com/Automattic/wp-calypso/pull/27020/files#diff-55fdd778b2cf9997c7ec135915fbca37L144

**Comment by sirreal** [comment]
This fails on CI eslint erroneously due to file deletion. #27076 contains a fix.

File: client/me/help/help-search/index.jsx (hunk 3 above)
Comment (sirreal) [inline client/me/help/help-search/index.jsx]: This isn't being fired, it's calling the action creator directly, not the prop 😛

Comment (sirreal) [reply client/me/help/help-search/index.jsx]: Fixed in 7c489d2bf1c7ceb11a596eac21385f60b1bfc711

File: client/state/help/reducer.js [hunk 4]
```diff
@@ -21,9 +20,26 @@ export const selectedSiteId = createReducer( null, {
...
 } );
 
+/**
+ * Responsible for the help search results links
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const links = ( state = [], action ) => {
```
Comment (sirreal) [inline client/state/help/reducer.js]: Initial state should be an empty object literal `{}`? `[]` isn't likely a problem, but also isn't the shape I'd expect.

File: client/me/help/help-search/index.jsx (hunk 1 above)
Comment (sirreal) [reply client/me/help/help-search/index.jsx]: I'd rather avoid a slew of additional `get` calls on render. A simple `fromApi: makeJsonSchemaParser( { type: 'object' } )` would ensure that it has the expected shape and we'll be able to access properties.

This isn't blocking and I know it's not introduced here.

**Review by sirreal — DISMISSED** [review]
This is testing well, thanks for the work here. I left a few notes, ship at will 👍

**Comment by sirreal** [comment]
It would be nice to get that CI passing before we ship 🙈

**Comment by sirreal** [comment]
Rebased to pull in eslint fix from #27076

**Review by sirreal — CHANGES_REQUESTED** [review]
The code seems to work well, however tests are failing and need to be updated.

**Comment by tyxla** [comment]
@sirreal thanks for the review and the eslint fix! I've fixed the tests - feel free to have another look.

File: client/state/help/reducer.js (hunk 4 above)
Comment (tyxla) [reply client/state/help/reducer.js]: 🤦‍♂️ of course. Fixed in 8ce4da1eca8d301b1dab2465c5f3f98ba520a981

File: client/me/help/help-search/index.jsx (hunk 3 above)
Comment (tyxla) [reply client/me/help/help-search/index.jsx]: 🤦‍♂️ second facepalm for the day. Thanks!

File: client/me/help/help-search/test/index.js [hunk 5]
```diff
@@ -18,8 +19,7 @@ import { HelpSearch } from '../';
 jest.mock( '', () => ( {} ) );
```
Comment (sirreal) [inline client/me/help/help-search/test/index.js]: 😕 What's going on here @ramonjd?

Let's please drop this if possible.

**Review by sirreal — APPROVED** [review]
Tests are passing now, manual testing looks good, nice work @tyxla!

Left a note that we can clean up later.

File: client/me/help/help-search/test/index.js (hunk 5 above)
Comment (ramonjd) [reply client/me/help/help-search/test/index.js]: Looks like a mistake. Thanks for picking it up. I'll fix.

**Pull request merged**